specs/0116-LLM-assisted-translations.md

```diff
@@ -0,0 +1,57 @@
+- Feature Name: LLM-Assisted Translations in Pontoon
+- Created: 2024-01-12
+- Associated Issue: #3064
+
+# Summary
+
+Introduce LLM-assisted translations in Pontoon to provide alternative translations, including formal and informal options, with the goal to enhance translation quality.
+
+# Motivation
+
+Modern Large Language Models (LLMs) — like OpenAI’s GPT-4 — have reached a quality level for localization tasks that is on par with dedicated machine translation engines. This feature is primarily focused on introducing a tool that aims to improve translation quality in Mozilla's projects. By incorporating LLM-assisted translations, Mozilla not only remains competitive with other Translation Management Systems (TMS) providers but also embraces new technologies to enhance the overall translation process. This feature also lays the groundwork for future integrations that could further enhance efficiency and broaden the scope of translation improvements within Mozilla's localization efforts.
+
+# Feature explanation
+
+In the Machinery tab of the Pontoon translate view, users will encounter a new drop-down option for each machine translation suggestion for a given string, initially offering three choices: 
+
+1) `MAKE INFORMAL` - Provide a more informal version of this translation 
+2) `MAKE FORMAL` - Provide a more formal version of this translation
+3) `REPHRASE` - Provide an alternative to this translation 
```

Comment:
Note: I think rephrase should be the first option, because I expect it to be the most common use case. Thoughts?

Reply:
Makes sense.

Reply:
Sounds good. I originally structured it this way to make the drop-down options go from longest phrase --> shortest phrase.